src/userdev/java/net/minecraftforge/gradle/userdev/DependencyManagementExtension.java

```diff
@@ -41,10 +50,40 @@ public Dependency deobf(Object dependency) {
         return deobf(dependency, null);
     }
 
-    public Dependency deobf(Object dependency, Closure<?> configure){
+    public Dependency deobf(Object dependency, Closure<?> configure) {
         Dependency baseDependency = project.getDependencies().create(dependency, configure);
         project.getConfigurations().getByName(UserDevPlugin.OBF).getDependencies().add(baseDependency);
 
         return remapper.remap(baseDependency);
     }
+
+    @SuppressWarnings({"ConstantConditions", "unchecked"})
+    public MavenPublication component(MavenPublication mavenPublication) {
+        project.getTasks().withType(GenerateModuleMetadata.class).forEach(generateModuleMetadata -> generateModuleMetadata.setEnabled(false));
+
+        mavenPublication.suppressAllPomMetadataWarnings(); //We have weird handling of stuff and things when it comes to versions and other features. No need to spam the log when that happens.
+
+        mavenPublication.pom(pom -> {
+            pom.withXml(xml -> {
+                final NodeList dependencies = MavenPomUtils.getDependenciesNodeList(xml);
+
+                final List<Node> dependenciesNodeList = (List<Node>) dependencies.stream()
+                        .filter(Node.class::isInstance)
+                        .map(Node.class::cast)
+                        .collect(Collectors.toList());
+
+                dependenciesNodeList.stream()
+                        .filter(el -> MavenPomUtils.hasChildWithText(el, MavenPomUtils.MAVEN_POM_NAMESPACE + "artifactId", "forge") && MavenPomUtils.hasChildWithText(el, MavenPomUtils.MAVEN_POM_NAMESPACE + "groupId", "net.minecraftforge"))
+                        .filter(el -> MavenPomUtils.hasChildWithText(el, MavenPomUtils.MAVEN_POM_NAMESPACE + "artifactId", "fmlonly") && MavenPomUtils.hasChildWithText(el, MavenPomUtils.MAVEN_POM_NAMESPACE + "groupId", "net.minecraftforge"))
+                        .forEach(el -> el.parent().remove(el));
```

Comment:
This is not gonna work. Has to be one filter call to be functional. `(artifactId == "fmlonly" || artifactId == "forge") && groupId == "net.mcf")`